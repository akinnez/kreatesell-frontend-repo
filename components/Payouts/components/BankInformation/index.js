import {Typography} from 'antd';
import styles from './index.module.scss';

const {Text} = Typography;

const BankInformation = ({bankDetails}) => (
	<>
		{bankDetails.country_id === '1' || bankDetails.country_id === '72' ? (
			<div
				className={`${styles.bank__info__details} ${styles.bank__info}`}
			>
				<p className={styles.text}>
					<Text>Currency:</Text>
					<Text>&nbsp; {bankDetails.currency || bankDetails.currency_name || 'NGN'}</Text>
				</p>
				<p className={styles.text}>
					<Text>Bank Name:</Text>
					<Text>&nbsp; {bankDetails.bank_name}</Text>
				</p>
				<p className={styles.text}>
					<Text>Account Number:</Text>
					<Text>&nbsp; {bankDetails.account_number}</Text>
				</p>
				<p className={styles.text}>
					<Text>Account Name:</Text>
					<Text>&nbsp; {bankDetails.account_name}</Text>
				</p>
			</div>
		) : (
			<div
				className={`${styles.bank__info__details} ${styles.paypal__info}`}
			>
				<p>
					<Text>PayPal Email:</Text>
					<Text>&nbsp; {bankDetails.account_name}</Text>
				</p>
			</div>
		)}
	</>
);

export default BankInformation;
