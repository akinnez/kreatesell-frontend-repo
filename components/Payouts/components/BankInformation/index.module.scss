@use "public/css/colors";
@use "public/css/mixins";

.bank__info__details > p {
  margin-bottom: 20px;
  @include mixins.flex(row, flex-start, center);
  flex-wrap: wrap;

  &:last-child {
    margin-bottom: 0;
  }

  > [class="ant-typography"] {
    font-weight: 500;
    font-size: 16px;

    &:first-child {
      color: colors.$gray-200;
    }

    &:last-child {
      color: colors.$black-100;
    }
  }
}

.bank__info > p > [class="ant-typography"]:first-child {
  width: 170px;
}

.paypal__info > p > [class="ant-typography"]:first-child {
  width: 130px;
}
