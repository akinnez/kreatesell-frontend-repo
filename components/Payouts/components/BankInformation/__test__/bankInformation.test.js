import BankInformation from "..";
import renderer from "react-test-renderer";

const mockProps = {
  bankDetails: {
    account_number: "02772727272",
    account_name: "Salvo Agency Intl",
    bank_name: "salvo Bank",
    currency: "NGN",
  },
};

const mockPropsTwo = (id) => ({
  ...mockProps,
  country_id: id,
});

describe("BankInformation: ", () => {
  it("renders a snapshot of <BankInformation /> when the country_id is neither 1 nor 72", () => {
    const bankInfo = renderer
      .create(<BankInformation {...mockProps} />)
      .toJSON();
    expect(bankInfo).toMatchSnapshot();
  });
  it("renders a snapshot of <BankInformation /> when the country_id is 1 ", () => {
    const bankInfo = renderer
      .create(<BankInformation {...mockPropsTwo("1")} />)
      .toJSON();
    expect(bankInfo).toMatchSnapshot();
  });
  it("renders a snapshot of <BankInformation /> when the country_id is 72 ", () => {
    const bankInfo = renderer
      .create(<BankInformation {...mockPropsTwo("72")} />)
      .toJSON();
    expect(bankInfo).toMatchSnapshot();
  });
});
