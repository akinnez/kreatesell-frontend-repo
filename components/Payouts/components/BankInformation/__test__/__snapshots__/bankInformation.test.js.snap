// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankInformation:  renders a snapshot of <BankInformation /> when the country_id is 1  1`] = `
<div
  className="bank__info__details paypal__info"
>
  <p>
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      PayPal Email:
    </span>
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
        
      Salvo Agency Intl
    </span>
  </p>
</div>
`;

exports[`BankInformation:  renders a snapshot of <BankInformation /> when the country_id is 72  1`] = `
<div
  className="bank__info__details paypal__info"
>
  <p>
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      PayPal Email:
    </span>
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
        
      Salvo Agency Intl
    </span>
  </p>
</div>
`;

exports[`BankInformation:  renders a snapshot of <BankInformation /> when the country_id is neither 1 nor 72 1`] = `
<div
  className="bank__info__details paypal__info"
>
  <p>
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      PayPal Email:
    </span>
    <span
      className="ant-typography"
      onClick={null}
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
        
      Salvo Agency Intl
    </span>
  </p>
</div>
`;
